pkg/build/api/v1beta3/types.go

```diff
@@ -167,6 +172,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStream tag or a DockerImage to
+	// copy source from
+	From kapi.ObjectReference `json:"from" description:"image reference"`
+
+	// Paths is a list of source and destination paths to copy from the image
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from
+	// a registry
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"pull secret for source image"`
```

Comment:
So we can do one of two things:
1) Add more keys to a single source secret to include an image pull secret.
2) Say that the current sourceSecret is really a git source secret and move it under the Git struct. 

I would vote for #2. Thoughts?